module/size-speed-range/foundry/ranges.js

```diff
@@ -0,0 +1,180 @@
+'use strict';
+
+import {displayMod} from '../../../lib/utilities.js';
+import {SSRT} from '../../../lib/gurps-foundry-ssrt-lib/src/js/SSRT.js';
+import {RangeBandTable} from '../domain/RangeBandTable.js';
+import convertToYards from '../../../lib/gurps-foundry-ssrt-lib/src/js/UnitsOfMeasure/Conversion.js';
+import parseDistance from '../../../lib/gurps-foundry-ssrt-lib/src/js/Parsing.js';
+import * as conversionMethods from '../../../lib/gurps-foundry-ssrt-lib/src/js/UnitsOfMeasure/Methods.js';
+
+const SYSTEM_NAME = 'gurps';
+const SETTING_NAME = 'rangeStrategy';
+const STRATEGY_STANDARD = 'Standard';
+const STRATEGY_RANGE_BANDS = 'Simplified';
+
+const getStrategy = () => game.settings.get(SYSTEM_NAME, SETTING_NAME);
+
+const simplifiedRangeTable = RangeBandTable.table().map(row => ({
+    moddesc: `${row.band} range (${row.maxDistanceYards} yds)`,
+    max: row.maxDistanceYards,
+    penalty: row.penalty,
+    desc: row.description
+}));
+
+const basicSetRangeTable = SSRT.speedRangeTable(16).map(row => ({
+    moddesc: `for speed/range ${row.yards} yds`,
+    max: row.yards,
+    penalty: row.penalty,
+    desc: `${row.yards} yds`
+}));
+// change the last entry description from 700 yards to 500+ yards
+basicSetRangeTable.forEach(row => {
+    if (row.penalty === -15) {
+        row.moddesc = 'for speed/range 500+ yds';
+        row.max = '500+';
+        row.desc = '500+ yds';
+    }
+});
+
+/*
+  Defines the range strategy used throughout the application. A range strategy
+  is defined as an ordered (closest range to farthest range) array of range 
+  bands. A range band is defined as a structure like this:
+
+  {
+    moddesc: <String: text to use in the modifier bucket>,
+    max: <num: number of yards that define the maximum distance of this band>,				
+    penalty: <num: modifier to use for ranged attacks in this band>,
+    desc: <String: text that describes the range>
+  }
+
+  Responsibilities:
+
+  - Defines a world setting to set the range strategy. Currently supported: 
+    * Standand (Size and Speed/Range Table from Basic).
+    * Simplifed (Range bands from Monster Hunters 2: The Enemy).
+   
+  - On update of the setting, update the modifier bucket and all actors.
+
+  - On start up (a 'ready' hook) set the range bands and modifiers based 
+    on the current setting value.
+
+  - On start up (a 'ready' hook) initializes the range ruler to show range penalties
+    according to the chosen strategy.
+    
+  - Maintains an instance variable (rangeTable) that contains the current set of
+    range bands based on the chosen strategy. This table is to be used only for display
+    and interactive roll mod application purposes
+
+  - Maintains an instance variable (bucketModifiers) that contains an array of
+    modifier text for the modifier bucket.
+ */
+export default class GURPSRange {
+    constructor() {
+        this._setup();
```

Comment:
I suggest adopting the convention to prefix the names of all private methods and class properties with _. This is what Foundry does, and is a useful thing to do in JS where there is no real private/public enforcement.

Reply:
Yep, I was doing that on the applydamage.js and damagecalculator.js.